packages/shared/src/open-feature.ts

```diff
@@ -12,23 +12,81 @@ import {
 import { isDefined } from './filter';
 import { BaseHook, EvaluationLifeCycle } from './hooks';
 import { DefaultLogger, Logger, ManageLogger, SafeLogger } from './logger';
-import { CommonProvider, ProviderMetadata, ProviderStatus } from './provider';
+import { ClientProviderStatus, CommonProvider, ProviderMetadata, ServerProviderStatus } from './provider';
 import { objectOrUndefined, stringOrUndefined } from './type-guards';
 import { Paradigm } from './types';
 
-export abstract class OpenFeatureCommonAPI<P extends CommonProvider = CommonProvider, H extends BaseHook = BaseHook>
-  implements Eventing, EvaluationLifeCycle<OpenFeatureCommonAPI<P>>, ManageLogger<OpenFeatureCommonAPI<P>>
+type AnyProviderStatus = ClientProviderStatus | ServerProviderStatus;
+
+/**
+ * A provider and its current status.
+ * For internal use only.
+ */
+export class ProviderWrapper<P extends CommonProvider<AnyProviderStatus>, S extends AnyProviderStatus> {
```

Comment:
This new class handles associating the provider and state, since the provider is now stateless. It's a much nicer solution than keeping a separate collection of statuses in the API object and trying to keep them in-sync with their associated provider.

We might be able to put additional logic in here as well.

It's exported from `/shared` since the SDKs need it, but not from the SDKs, themselves... I think this is OK.